File: Assets/Code/Logic/Item/Model/ItemData.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;
using Common.Util;
using Logic.Enums;

namespace Logic.Item.Model
{
	public class ItemData 
	{
		private static Dictionary<int, ItemData> _itemDataDictionary;
		
		public static Dictionary<int, ItemData> GetItemDatas()
		{
			if (_itemDataDictionary == null)
			{
				_itemDataDictionary = CSVUtil.Parse<int, ItemData>("config/csv/item", "id");
			}
			return _itemDataDictionary;
		}
		
		public static Dictionary<int, ItemData> ItemDataDictionary
		{
			get
			{
				if (_itemDataDictionary == null)
				{
					GetItemDatas();
				}
				return _itemDataDictionary;
			}
		}
		public static ItemData GetItemDataByID (int itemtDataID)
		{
			ItemData equipmentData = null;
			if (ItemDataDictionary.ContainsKey(itemtDataID))
			{
				equipmentData = ItemDataDictionary[itemtDataID];
			}
			return equipmentData;
		}
		//基础资源类型
		public static ItemData GetBasicResItemByType(BaseResType type)
		{
			ItemData data;
			foreach(var value in ItemDataDictionary)
			{
				data = value.Value;
				if(data.type == (int)type)
					return data;
			}
			return null;
		}
		[CSVElement("id")]
		public int id;

		[CSVElement("type")]
		public int type;

		[CSVElement("name")]
		public string name;

		[CSVElement("icon")]
		public string icon;

		[CSVElement("des")]
		public string des;

		public ItemQuality itemQuality = ItemQuality.White;
		[CSVElement("quality")]
		public int quality
		{
			set
			{
				itemQuality = (ItemQuality)value;
			}
		}

		[CSVElement("money")]
		public int money;

		[CSVElement("prize_id")]
		public string prize_id;

		[CSVElement("jump_page")]
		public string jump_page;


		public List<RoleType> heroTypeList = new List<RoleType>();

		[CSVElement("hero_type")]
		public string hero_type
		{
			set
			{
				string[] types = value.Split(CSVUtil.SYMBOL_SEMICOLON);
				for(int i = 0,count = types.Length;i<count;i++)
				{
					int type = types[i].ToInt32();
					if( type != 0 )
						heroTypeList.Add((RoleType)(types[i].ToInt32()));
				}

			}
		}

		[CSVElement("count")]
		public int sort;	//在背包中的排序

		public int universal_id;
		[CSVElement("universal_id")]
		public string universal_idString
		{
			set
			{
				if(!string.IsNullOrEmpty(value))
					universal_id = value.ToInt32();
			}
		}
	}

}

File: Assets/ToLua/Lua/user/ui/item/model/item_data.lua
local t = {}
t.data = {}
local item = {}
item.__index = item

function item.New(table)
	local o = {}
	setmetatable(o,item)
	o.id = tonumber(table.id)
	o.type = tonumber(table.type)
	o.name = table.name
	o.icon = table.icon
	o.des = table.des
	o.quality = tonumber(table.quality)
  o.lv = tonumber(table.lv)
	o.money = tonumber(table.money)
	o.prize_id = table.prize_id
	o.stack = table.stack
	o.jump_page = table.jump_page
	o.hero_type = string.split(table.hero_type,';')
	o.sort = tonumber(table.count)
	o.universal_id = tonumber(table.universal_id)
  
	return o
end

function item:IconPath()
  return 'sprite/item_icon/'..self.icon
end

function t.GetDataById(id)
	if t.data[id] then
		return t.data[id]
	else
		Debugger.LogError(string.format('can not get item id from config:%s',id))
	end
end

function t.GetBasicResItemByType(baseResType)
  for k, v in pairs(t.data) do
    if v.type == baseResType then
      return v
    end
  end
  return nil
end  

local function Start()
	local origin = dofile('item')
  print('item origin ',origin)
	origin.ForEach(function(id,table)
		local newItem = item.New(table)
		t.data[id] = newItem
	end)
end

Start()
return t